components/cdc/src/service.rs

```diff
@@ -473,14 +490,121 @@ impl Service {
             }
         });
 
+        // Create atomic variable to track last flush time
+        let last_flush_time = Arc::new(AtomicCell::new(Instant::now()));
+        let last_flush_time_for_forward = last_flush_time.clone();
+        let last_flush_time_for_watchdog = last_flush_time.clone();
+        let peer_for_watchdog = ctx.peer().to_string();
+
         let peer = ctx.peer();
+        let (cancel_tx, mut cancel_rx) = tokio::sync::oneshot::channel::<()>();
+
+        // Create cancelCh for eventDrain.forward exit signal
+        let (forward_exit_tx, forward_exit_rx) = tokio::sync::oneshot::channel::<()>();
```

Comment:
It looks that `tikv_util::future::paired_future_callback();` can be used to send signal, it's used in the tikv-cdc.